tests/test_grains_minicircle.py

```diff
@@ -105,6 +105,25 @@ def test_remove_small_objects(
     return fig
 
 
+@pytest.mark.mpl_image_compare(baseline_dir="resources/img/")
+def test_area_thresholding(
+    minicircle_area_thresholding: Grains, plotting_config: dict, plot_dict: dict, tmp_path
+) -> None:
+    """Test removal of small objects via absolute area thresholding."""
+    plotting_config["type"] = "binary"
+    assert isinstance(minicircle_area_thresholding.directions["upper"]["removed_small_objects"], np.ndarray)
+    assert minicircle_area_thresholding.directions["upper"]["removed_small_objects"].shape == (1024, 1024)
+    assert minicircle_area_thresholding.directions["upper"]["removed_small_objects"].sum() == 42378
+    plotting_config = {**plotting_config, **plot_dict["removed_small_objects"]}
+    fig, _ = Images(
+        data=minicircle_area_thresholding.directions["upper"]["removed_small_objects"],
+        output_dir=tmp_path,
+        pixel_to_nm_scaling_factor=minicircle_area_thresholding.pixel_to_nm_scaling,
+        **plotting_config,
+    ).plot_and_save()
+    return fig
```

Comment:
Could you add the image to the repository and commit it please?

I think it should be something like...

```
git add tests/resources/img/test_area_thresholding.png
```

...assuming you've generated it with the `pytest tests/test_grains_minicircle.py --mpl-generate-path=tests/resources/img`.

Reply:
> > After meeting, this area replacement would work well as a factory method where values are specified and if not, the already present method using OTSU should be done
> 
> If I've understood the code correctly in this PR the area threshold being used is based on a bounding box. The current methods of thresholding based on size are, and hopefully @SylviaWhittle can correct me if I'm wrong, based on the number of connected pixels.
> 
> Whilst both approaches will result in a number (one an area, the other the number of connected pixels) their type and meaning are subtly different, to which end careful consideration of whether this should be included in the current thresholding functions (/factory method) should be given.

So this method counts the number of pixels each grain has and multiplies it by `px_to_nm_scaling ** 2` to obtain an area, then it uses this to threshold. However, I've just realised I could go the other way and reuse the `remove_noise` function to do this too instead of the one I made.

@SylviaWhittle approach uses `remove_small_objects` which first calculates a `minimum_grain_size` via finding grains which are greater than a threshold obtained via otsu (i.e. reduces grain area list to those of bigger objects). Then this `minimum _grain_size is calculated by `np.median(grain_areas) - (1.5 * (np.quantile(grain_areas, 0.75) - np.quantile(grain_areas, 0.25)))` and finally `remove_small_objects` removes objects < this value. Let me know if I've got that right?

Reply:
Actually, skimage's remove small objects func can't remove objects larger than a value nor is there a similar func to do so so I'll keep mine

Reply:
> > > After meeting, this area replacement would work well as a factory method where values are specified and if not, the already present method using OTSU should be done
> > 
> > 
> > If I've understood the code correctly in this PR the area threshold being used is based on a bounding box. The current methods of thresholding based on size are, and hopefully @SylviaWhittle can correct me if I'm wrong, based on the number of connected pixels.
> > Whilst both approaches will result in a number (one an area, the other the number of connected pixels) their type and meaning are subtly different, to which end careful consideration of whether this should be included in the current thresholding functions (/factory method) should be given.
> 
> So this method counts the number of pixels each grain has and multiplies it by `px_to_nm_scaling ** 2` to obtain an area, then it uses this to threshold. However, I've just realised I could go the other way and reuse the `remove_noise` function to do this too instead of the one I made.
> 
> @SylviaWhittle approach uses `remove_small_objects` which first calculates a `minimum_grain_size` via finding grains which are greater than a threshold obtained via otsu (i.e. reduces grain area list to those of bigger objects). Then this `minimum _grain_size is calculated by `np.median(grain_areas) - (1.5 * (np.quantile(grain_areas, 0.75) - np.quantile(grain_areas, 0.25)))`and finally`remove_small_objects` removes objects < this value. Let me know if I've got that right?

Correct. But it can't remove larger objects, only smaller ones. I think we might want to shift to just using our own custom Python function rather than using an incomplete (can't do larger grains) version from scikit?